### ISSUE TITLE
Code Helper (Renderer) 100% CPU usage - tsserver

### ISSUE THREAD

**Issue by vobarian** [microsoft/vscode#111683]
Issue Type: <b>Performance Issue</b>

After working for a while in a TypeScript/React project, I frequently find a process in Activity Monitor named Code Helper (Renderer) using 100% CPU. There is no specific step to reproduce this, it just happens after a while doing normal code editing. It might be related to using Cmd+Click "hyperlink" style code navigation but I haven't confirmed that. All extensions are disabled. VS Code's process explorer shows 100% CPU on electron_node tsserver.js. Once this happens it will remain at 100% CPU usage until I close the window.

VS Code version: Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f, 2020-11-11T01:11:34.018Z)
OS version: Darwin x64 19.6.0

<details>

<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 3|
|Memory (System)|16.00GB (0.30GB free)|
|Process Argv|--crash-reporter-id bc101f13-7f74-4229-8f6d-b55ef617e9a9|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
   13	   147	  3132	code main
    2	   115	  3135	   gpu-process
    0	    33	  3137	   utility
    0	   328	  3140	   window (StickyHeader.tsx — cfo-common)
    0	     0	  3217	     /bin/bash -l
    0	    82	  3260	     extensionHost
    0	    66	  3297	       electron_node tsserver.js 
   97	   229	  3298	       electron_node tsserver.js 
    0	    49	  3309	         electron_node typingsInstaller.js typesMap.js 
    0	    33	  3326	       /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=3260
    0	    33	  6041	       /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/css-language-features/server/dist/node/cssServerMain --node-ipc --clientProcessId=3260
    0	    66	  3322	     watcherService
    0	    49	 12463	     searchService
    0	    98	  3313	   shared-process
    0	     0	 13147	     /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	   377	 12850	   window (undefined)
    0	    66	 13144	   window (Issue Reporter)
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (StickyHeader.tsx — cfo-common)
|    Folder (cfo-common): 12149 files
|      File types: gzip(10790) tsx(268) ts(205) scss(185) svg(136) js(95)
|                  png(77) jpg(58) json(21) gz(13)
|      Conf files: package.json(5) tsconfig.json(5) launch.json(1)
|                  settings.json(1) tslint.json(1)
|      Launch Configs: chrome(2);
```

</details>
Extensions: none

[CPU-20201201T161402.cpuprofile.zip](https://github.com/microsoft/vscode/files/5625828/CPU-20201201T161402.cpuprofile.zip)

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

**Comment by vobarian** [comment]
I think I have narrowed down the steps to reproduce the issue. It seems to happen when I run `npm install` while the project is open in VS Code. Our project uses Lerna and our package.json scripts do this when install is run: `lerna bootstrap --hoist --nohoist=@carfax/*`. Perhaps it has something to do with the symlinks that Lerna creates?

I have **not** been able to reproduce it yet in this Insiders build:

> Version: 1.52.0-insider
Commit: f4187f22026ff6537f2ca8132ad3209fceb5071f
Date: 2020-12-02T13:18:35.551Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.6.0

**Comment by mjbvz** [comment]
Ok, if you can't reproduce this in out insiders build, then the issue may have already been fixed. The current insiders build will become the next stable VS Code release

If you do see this in VS Code insiders, please let me know and I can share steps on how to collect logs so that we can investigate this

**Comment by vobarian** [comment]
@mjbvz When I tested this in the Insiders build on 12/2, it was not reproducing the issue. Since that time, VS Code 1.52.0 came out and now 1.52.1. Unfortunately the issue is still occurring in 1.52.1 but when I downloaded the latest Insiders build today (4a875e23d20b64504a818834f3fa4c40adb8d480) it is again not occurring.

So the assumption that it had been fixed in the Insiders build and would be fixed in the next stable release seems to not be true. Instead there is some persistent difference between the stable build and the Insiders build.

Again, when I have our TypeScript/React project open in VS Code and run "npm install", CPU usage goes to 100% and remains there until I close the window. This happens regardless of whether I run the command from the embedded Terminal window or from a completely separate external terminal. VS Code Process Explorer shows the offending process to be "electron_node tsserver.js"

I have uninstalled all extensions and restarted Code. Additionally I completely deleted it from my Applications folder and downloaded and installed a fresh copy of the stable release:

> Version: 1.52.1
> Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
> Date: 2020-12-16T16:30:02.420Z
> Electron: 9.3.5
> Chrome: 83.0.4103.122
> Node.js: 12.14.1
> V8: 8.3.110.13-electron.0
> OS: Darwin x64 19.6.0
